### PR TITLE
tgui-next: Infrared emitter & Proximity Sensor

### PR DESCRIPTION
**Pull request by Arkatos1**
## About The Pull Request

This PR converts last two remaining assemblies to tgui-next, namely Infrared emitter and Proximity sensor.

<details>
  <summary>Old Infrared emitter UI</summary>

InfraredOLD

</details>

New Infrared emitter UI:

InfraredNEW

<details>
  <summary>Old Proximity sensor UI</summary>

ProxOLD

</details>

New Proximity sensor UI:

ProxNEW

## Why It's Good For The Game

Better UI.

## Changelog
:cl: Arkatos
add: Infrared emitter now uses tgui-next.
add: Proximity sensor now uses tgui-next.
/:cl:

### CODE REVIEW

File: code/modules/assembly/proximity.dm [hunk 1]
```diff
@@ -114,49 +110,48 @@
...
+/obj/item/assembly/prox_sensor/ui_act(action, params)
+	if(..())
 		return
 
-	if(usr)
-		attack_self(usr)
+	switch(action)
+		if("scanning")
+			var/toggle = scanning ? FALSE : TRUE
+			toggle_scan(toggle)
```
Comment (4dplanner) [inline code/modules/assembly/proximity.dm]: Is this not just `toggle_scan(!scanning)`?

Comment (Arkatos1) [reply code/modules/assembly/proximity.dm]: Yep, you are right. Good catch.

File: code/modules/assembly/proximity.dm [hunk 2]
```diff
@@ -114,49 +110,48 @@
...
+				sensitivity_change(value)
+				. = TRUE
+		if("time")
+			timing = !timing
+			update_icon()
+			. = TRUE
+		if("input")
+			var/value = text2num(params["adjust"])
+			if(value)
+				time = min(max(round(time+value), 0), 600)
```
Comment (4dplanner) [inline code/modules/assembly/proximity.dm]: perhaps use CLAMP?

Comment (Arkatos1) [reply code/modules/assembly/proximity.dm]: Done.

**Review by stylemistake — DISMISSED** [review]

**Comment by Cyberboss** [comment]
Rebuild

**Comment by Arkatos1** [comment]
@Cyberboss Done.

**Comment by PKPenguin321** [comment]
just curious, do infrared emitters actually work or are they still as broken as ever?

**Comment by Arkatos1** [comment]
Backend is the same, so any issues with these assemblies will remain, unless problem was in the interface itself.

**Comment by cacogen** [comment]
these are never used

**Review by kriskog — APPROVED** [review]

**Pull request merged**